File: autonomie/static/js/date.js
/*
 * * Copyright (C) 2012-2013 Croissance Commune
 * * Authors:
 *       * Arezki Feth <f.a@majerti.fr>;
 *       * Miotte Julien <j.m@majerti.fr>;
 *       * Pettier Gabriel;
 *       * TJEBBES Gaston <g.t@majerti.fr>
 *
 * This file is part of Autonomie : Progiciel de gestion de CAE.
 *
 *    Autonomie is free software: you can redistribute it and/or modify
 *    it under the terms of the GNU General Public License as published by
 *    the Free Software Foundation, either version 3 of the License, or
 *    (at your option) any later version.
 *
 *    Autonomie is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *    GNU General Public License for more details.
 *
 *    You should have received a copy of the GNU General Public License
 *    along with Autonomie.  If not, see <http://www.gnu.org/licenses/>.
 */


function getOneMonthAgo(){
  var today = new Date();
  var year = today.getUTCFullYear();
  var month = today.getUTCMonth() - 1;
  var day = today.getUTCDate();
  return new Date(year, month, day);
}

function parseDate(isoDate){
  /*
   * Returns a js Date object from an iso formatted string
   */
   var splitted = isoDate.split('-');
   var year = parseInt(splitted[0], 10);
   var month = parseInt(splitted[1], 10) - 1;
   var day = parseInt(splitted[2], 10);
   return new Date(year, month, day);
}
function formatPaymentDate(isoDate){
  /*
   *  format a date from iso to display format
   */
  if ((isoDate !== '') && (isoDate !== null)){
    return $.datepicker.formatDate("dd/mm/yy", parseDate(isoDate));
  }else{
    return "";
  }
}
var formatDate = formatPaymentDate;
function getDateFromIso(isoDateStr, dateFormat){
 var strdate = "";
 if (isoDateStr !== ''){
  var date = parseDate(isoDateStr);
  strdate = $.datepicker.formatDate(dateFormat, date);
 }
 return strdate;
}

function getTimeFromIso(isoTimeStr, timeFormat){
  var splitted = isoTimeStr.split(':');
  var hour = parseInt(splitted[0], 10);
  var min = parseInt(splitted[1], 10);
  return $.datepicker.formatTime(timeFormat, {hour: hour, minute: min});
}

function getDateTimeFromIso(isoStr, dateFormat, timeFormat){
  var strdatetime = "";
  if (isoStr !== ''){
    var isoDateStr = isoStr.split(' ')[0];
    var isoTimeStr = isoStr.split(' ')[1];
    var strdate = getDateFromIso(isoDateStr, dateFormat);
    var strTime = getTimeFromIso(isoTimeStr, timeFormat);
    strdatetime = strdate + " " + strTime;
  }
  return strdatetime;
}
function dateToIso(dateObject){
    var year = dateObject.getFullYear();
    var month = dateObject.getMonth()+1;
    var dt = dateObject.getDate();

    if (dt < 10) {
        dt = '0' + dt;
    }
    if (month < 10) {
        month = '0' + month;
    }
    return year + "-" + month + "-" + dt;
}

File: autonomie/utils/date.py
# -*- coding: utf-8 -*-
# * Copyright (C) 2012-2014 Croissance Commune
# * Authors:
#       * Arezki Feth <f.a@majerti.fr>;
#       * Miotte Julien <j.m@majerti.fr>;
#       * TJEBBES Gaston <g.t@majerti.fr>
#
# This file is part of Autonomie : Progiciel de gestion de CAE.
#
#    Autonomie is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    Autonomie is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with Autonomie.  If not, see <http://www.gnu.org/licenses/>.
"""
Date manipulation utilities
"""
import datetime


def str_to_date(str_date):
    """
    Transform a date string to a date object
    """
    res = ""
    if str_date is not None:
        try:
            res = datetime.datetime.strptime(str_date, "%d/%m/%Y")
        except ValueError:
            try:
                res = datetime.datetime.strptime(str_date, "%d-%m-%Y")
            except ValueError:
                res = ""
    return res


def format_short_date(date):
    """
        return a short printable version of the date obj
    """
    if isinstance(date, datetime.date):
        resp = date.strftime("%e/%m/%Y")
    elif not date:
        resp = u""
    else:
        resp = datetime.datetime.fromtimestamp(float(date)).strftime(
                                                            "%d/%m/%Y %H:%M")
    return resp


def format_datetime(datetime_object, timeonly=False):
    """
    format a datetime object
    """
    res = datetime_object.strftime("%H:%M")
    if not timeonly:
        day = datetime_object.strftime("%d/%m/%Y")
        res = u"%s à %s" % (day, res)
    return res


def format_long_date(date):
    """
        return a long printable version of the date obj
    """
    if isinstance(date, datetime.date):
        resp = u"{0}".format(
            date.strftime("%e %B %Y").decode('utf-8').capitalize()
        )
    elif not date:
        resp = u""
    else:
        date = datetime.datetime.fromtimestamp(float(date))
        resp = u"{0}".format(
            date.strftime("%e %B %Y").decode('utf-8').capitalize()
        )
    return resp


def format_date(date, short=True):
    """
        return a pretty print version of the date object
    """
    if short:
        return format_short_date(date)
    else:
        return format_long_date(date)
